src/picongpu/include/plugins/output/images/PngCreator.tpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright 2013-2016 Axel Huebl, Heiko Burau, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pmacc_types.hpp"
+#include "simulation_defines.hpp"
+#include "plugins/output/images/PngCreator.hpp"
+#include "memory/boxes/DataBox.hpp"
+#include "plugins/output/header/MessageHeader.hpp"
+#include "verify.hpp"
+
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+
+#if( PIC_ENABLE_PNG == 1 )
+#   include <pngwriter.h>
+#endif
+
+namespace picongpu
+{
+    template< class Box >
+    inline void PngCreator::createImage(
+        const Box data,
```

Comment:
I realized that `createImage` is a host only function without any prefix, therefore we should not use an self implemented ignore function. For this case we should use `boost::ignore_unused`.
Never the less we need a equivalent function with the prefix `__host__ __device__` for other code parts.